libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -143,6 +143,8 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> {
     public async beginDialog(dc: DialogContext, options?: O): Promise<DialogTurnResult> {
         this.onPushScopedServices(dc.context);
         try {
+            await this.checkForChanges(dc);
```

Comment:
>    await this.checkForChanges(dc); [](start = 8, length = 35)

Should checking for changes be built into the dialog context itself?  It's the thing that is orchestrating switching between dialogs isn't it?